src/cmake-tools.ts

```diff
@@ -484,8 +514,13 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
    * Implementation of `cmake.build`
    */
   async build(target_?: string): Promise<number> {
-    // First, reconfigure if necessary
-    if (await this._needsReconfigure()) {
+    // First, save open files
+    if (!await this.maybeAutoSaveAll()) {
+      return -1;
```

Comment:
Is this error code unique and is it needed?

Reply:
Not unique, but its purpose is to signal to the caller that the build "failed" (because saving files failed).